client/containers/DashboardEdges/NewEdgeInput.js

```diff
@@ -0,0 +1,93 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { MenuItem, Position } from '@blueprintjs/core';
+import { Suggest } from '@blueprintjs/select';
+import isUrl from 'is-url';
+
+import { PubMenuItem } from 'components';
+import { useThrottled } from 'utils/hooks';
+import { fuzzyMatchPub } from 'utils/fuzzyMatch';
+
+require('./newEdgeInput.scss');
+
+const propTypes = {
+	availablePubs: PropTypes.arrayOf(
+		PropTypes.shape({
+			title: PropTypes.string,
+			avatar: PropTypes.string,
+		}),
+	).isRequired,
+	onSelectItem: PropTypes.func.isRequired,
+	usedPubIds: PropTypes.arrayOf(PropTypes.string).isRequired,
+};
+const defaultProps = {};
+
+const renderInputValue = () => '';
+
+const NewEdgeInput = (props) => {
+	const { availablePubs, usedPubIds, onSelectItem } = props;
+	const [queryValue, setQueryValue] = useState('');
+	const [suggestedItems, setSuggestedItems] = useState([]);
+	const throttledQueryValue = useThrottled(queryValue, 250, true, true);
+
+	useEffect(() => {
+		if (isUrl(queryValue)) {
+			setSuggestedItems([]);
+		} else if (queryValue) {
+			setSuggestedItems(
+				availablePubs
+					.filter((pub) => fuzzyMatchPub(pub, queryValue) && !usedPubIds.includes(pub.id))
+					.slice(0, 5)
+					.map((pub) => ({ type: 'pub', pub: pub })),
+			);
+		} else {
+			setSuggestedItems([]);
+		}
+	}, [availablePubs, queryValue, throttledQueryValue, usedPubIds]);
```

Comment:
Should we use `throttledQueryValue` in this effect and remove `queryValue` from the array of dependencies?

Reply:
Also - does `suggestedItems` need to be a state field or can it be derived from the query value each update?

Reply:
For now, yeas — but things will get more complicated when we start making asynchronous requests for suggested items.

Reply:
++ to using `throttledQueryValue`, though.